Core/include/Acts/Utilities/BinAdjustment.hpp

```diff
@@ -141,12 +192,17 @@ BinUtility adjustBinUtility(const BinUtility& bu, const Surface& surface) {
     auto rBounds = dynamic_cast<const RadialBounds*>(&(surface.bounds()));
     // Return specific adjustment
     return adjustBinUtility(bu, *rBounds, surface.transform(GeometryContext()));
-  }
-
-  throw std::invalid_argument(
-      "Bin adjustment not implemented for this surface yet!");
+  } else if (surface.type() == Surface::Plane) {
```

Comment:
Planes can also have different bounds than just `RectangleBounds` - this needs to be better protected.
The cast could fail and the dereferencing would do its (bad) deed to it.